server/drivers/bigquery/index.js

```diff
@@ -0,0 +1,144 @@
+const { BigQuery } = require('@google-cloud/bigquery');
+const { formatSchemaQueryResults } = require('../utils');
+
+const id = 'bigquery';
+const name = 'BigQuery';
+
+/**
+ * Run query for connection
+ * Should return { rows, incomplete }
+ * @param {string} queryString
+ * @param {object} connection
+ */
+function runQuery(queryString, connection) {
+  const bigquery = new BigQuery({
+    projectId: connection.projectId || process.env.GCP_PROJECT,
+    keyFilename: connection.keyFile || process.env.GOOGLE_CREDENTIALS_FILE
+  });
+
+  let incomplete = false;
+
+  const query = {
+    query: queryString,
+    // Location must match that of the dataset(s) referenced in the query.
+    location: connection.datasetLocation
+  };
+
+  // TODO: should maxRows apply to non-SELECT statements?
+  return bigquery
+    .createQueryJob(query)
+    .then(([job]) => {
+      // Waits for the query to finish
+      return job.getQueryResults({
+        maxResults: connection.maxRows + 1
```

Comment:
What is the `+ 1` for?

Reply:
It's so the code can tell if the query would return more results than `maxRows`. I haven't explored that, but I guess it's a SQLPad feature to have that limit and be able to tell the user if the results were truncated or not.